### ISSUE TITLE
"Limit To" Alert Bug

### ISSUE THREAD

**Issue by lukas-adoma** [DependencyTrack/dependency-track#1150]
### Current Behavior:
When a vulnerable component from project one is added to project two, project one receives a notification saying "New Vulnerability Identified". This occurs even when project one notifications have "Limit To" set to project one.

### Steps to Reproduce:
Notification set to portfolio, information, Microsoft Teams. Limit notifications to project one containing a component with vulnerabilities. Add the vulnerable component from project one to project two using the "Add Component" button.

### Expected Behavior:
Project two is unrelated to project one, so there should be no notification for project one when project two's components are changed.

### Environment:
  
  - Dependency-Track Version: v4.3.2
  - Distribution: [ Docker ]
  - BOM Format & Version: CycloneDX v1.2
  - Database Server: [ H2 | MSSQL | MySQL | PostgreSQL ]
  - Browser: Google Chrome

### Additional Details:
This issue would even cause the same notification to be sent twice.

**Comment by DrakezulsMinimalism** [comment]
I also just experienced the same for "PROJECT_AUDIT_CHANGE" where:
- an alert notification with "AUDIT CHANGE + NEW VULN + NEW VULN DEP" exists with LIMIT TO project A1, A2, A3
- the same combination exists with LIMIT TO project B1 and B2 [different recipients than then A-projects]
- Project B1 received audit comments
- recipients of project A's alert received a mail

Also, all the information in the mail was empty:
```
Violation Analysis Decision: Marking Finding as NOT SET

--------------------------------------------------------------------------------

Analysis Type:  Project Analysis
Analysis State:    
Suppressed:        
Vulnerability ID:  
Vulnerability URL: /vulnerability/?source=&vulnId=
Severity:          
Source:            
Component:         
Component URL:     /component/?uuid=
Project:           
Project URL:       /project/?uuid=

--------------------------------------------------------------------------------

An violation analysis decision was made to a policy violation affecting a project
```

**Comment by DrakezulsMinimalism** [comment]
@stevespringett I upgraded DependencyTrack to 4.4.1 but this bug still happened to me:
I have a notification setup for multiple "projects" of group B with "Vuln + Audit change".
I also have a notification setup for multiple "project" of group A with ONLY vulnerabilities.

I made an audit change to project A (after receiving a notification about a detected vulnerability with correct recipients) which triggered the notification of group B recipients.
